src/ProductPage.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import css from './ProductPage.module.css';
+
+interface Product {
+  category: {
```

Comment:
It will be better to split this into a new interface.
```suggestion
interface Category {
  id: number,
  name: string,
  image: string
 };
```